app/controllers/DashboardController.scala

```diff
@@ -118,10 +98,32 @@ trait DashboardController extends FrontendController with GatekeeperAuthWrapper
         approval = lastApproval(app)
         submission <- lastSubmission(app)
         admins <- administrators(app)
-        approvedApp = application(app.application, approval, admins, submission)
+        approvedApp: ApprovedApplication = application(app.application, approval, admins, submission)
       } yield Ok(approved(approvedApp))
   }
 
+  private def lastSubmission(app: ApplicationWithHistory)(implicit hc: HeaderCarrier): Future[SubmissionDetails] = {
+    val submission: StateHistory = app.history.filter(_.state == State.PENDING_GATEKEEPER_APPROVAL)
+      .sortWith(StateHistory.ascendingDateForAppId)
+      .lastOption.getOrElse(throw new InconsistentDataState("pending gatekeeper approval state history item not found"))
+
+    developerConnector.fetchByEmail(submission.actor.id).map(s =>
+      SubmissionDetails(s"${s.firstName} ${s.lastName}", s.email, submission.changedAt))
+  }
```

Comment:
This seems like the kind of thing that should have always been in a service - controllers should (imo, and ideally) not be doing lots of data transformation and calling multiple connectors..